### DISCUSSION TITLE
Installing homebrew and echo password

### DISCUSSION THREAD

**Discussion by RHamers** [Homebrew/discussions#60] [Everyday usage] [answered]
Hi everyone,

I have a question regarding installing homebrew with the following command:
`/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)" `

This script prompts the user for his (admin)password but I want to automate this since I want to automatically install homebrew as a dependency to my software. (I'm using C#+Mono and I start a process through System.Diagnostics.Process)

How can I send my password through STDIN (or something) to the prompt of the script. "
I tried several commands like:
`/bin/bash -c "echo {password} | $(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh) " `
or
`/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)  <<< {password}" `
or
`/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)  <<STDIN\n{password}\nSTDIN\n" `
or
`/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)" 
 <<STDIN\n{password}\nSTDIN\n `

But the above commands do not work, it still prompts me for my password. 
Could anyone help me with this? 👍 

Thanks

**Comment by reitermarkus** [comment, marked as answer]
You can use `SUDO_ASKPASS` for this.

**Reply by RHamers** [reply]
how exactly ? :)

**Reply by jonchang** [reply]
Look at `man sudo` or https://www.sudo.ws/man/1.8.31/sudo.man.html#A for the `askpass` option.

**Reply by RHamers** [reply]
Ok, thanks :) I will update the thread once I get it working.

**Reply by RHamers** [reply]
Hey,

I have fixed it yet it feels like one big hack to me, let me explain:
*note this is in C# which explains the string interpolation.

I first add my sudo password to the macos keychain with the following command
`string addPasswordCommand = $"echo {osPassword} | sudo -S security add-generic-password -a {GetOsUserName()} -s {environment} -w {osPassword}";`

Next I run:

`string testCommand = $"SUDO_ASKPASS=$(security find-generic-password -a {GetOsUserName()} -s {environment} -w) /bin/bash -c \"echo \"{osPassword}\" | sudo -S echo \"hi\"\r$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)\"";`

and lastly, I simulate the user pressing <return> on his keyboard to confirm homebrew install by running 
`\r`

It works but it still does not feel right. ( but I can start refactoring now)

Thanks

**Reply by Cyberbeni** [reply]
`export CI=true` and you can skip the enter pressing
You should also use the `-k` option for sudo, and maybe also add `-p ""`

**Reply by RHamers** [reply]
Thx, I will look into it soon and update the thread accordingly ;)
